Афигеть шип тест ивент

KIBORG04 opened the pull request:
<!-- Текст между стрелками - это комментарии - они не будут видны в вашем PR. -->

## Описание PR <!-- Опишите здесь ваш Pull Request. Что он изменяет? На что еще это может повлиять? -->

Основные фичи:
 - Шип-тест команда имеет имена-сикеи
 - Можно изменить название шаттла (grid'а)
 - Измененные РПГ-шки с разным уроном, боезапасом и скоростью стрельбой.
 - Админ ивент для моментального спавна 2х кораблей (@Dushess0 нужно у `Content.Server/StationEvents/Events/Theta/ShipEvent.cs` прописывать пути до корабликов, мне лень было делать без хардкода)

Остальноена скриншотах и видосиках в дисе.

Для началаивента:

Супер-способ:
 - Выбрать любую карту
 - Попросить всех людей зайти в госты
 - Ввести в консоль addgamerule ShipEvent (мб где-то ошибся писал по памяти)
 - Профит

<details> 
  <summary>Сложные способы</summary>
  
Способ со спавном на другом слое и потом лететь на слой станции. Типа чтобы челы подготовились
 - Выбрать любую карту.
 - Попросить всех людей зайти в госты.
 - В консоль `loadmap 2 /Maps/Shuttles/ship_test_1.yml`, где 2 постоянно инкрементить с новым шаттлом
 - В консоль `tp 0 0 2`, чтобы телепортироваться туда, где 2 тот же номер, что и у комманды выше
 - Запустить гостов. На 1 метке 5 спавнов.
 
Способ со спавном сразу на слое станции
 - Выбрать любую карту.
 - Попросить всех людей зайти в госты.
 - В консоль `loadgrid 1 /Maps/Shuttles/ship_test_1.yml 390 -390`, где цифры в конце - x и y координаты вашего госта.
 Координаты находятся в F3 в :

![image](https://user-images.githubusercontent.com/26389417/204336687-83a74a23-c3b4-47c2-b8d9-be490366174a.png)

 - Запустить гостов. На 1 метке 5 спавнов.

</details>

KIBORG04 committed:
wow i did it

KIBORG04 committed:
map

KIBORG04 committed:
ok

KIBORG04 commented on Resources/Prototypes/Damage/modifier_sets.yml:
```diff
@@ -110,6 +110,19 @@
     Heat: 0.9
     Poison: 0.9
 
+- type: damageModifierSet
```
оно кстати не иммортал и хз в каких случаях все равно дамажится, это даже лучше

Dushess0 replied:
не знаю задумано или нет, но у меня вышло сдохнуть получил взрыв от фуелтанка и РПГ

KIBORG04 replied:
Я понятия не имею как работает урон, как сделать бессмертным и почему ГОДМОД вообще работает. Но эта штука впринципе защищает от космоса, от пуль, от милли, что еще надо?

LudwigVonChesterfield commented on Content.Server/Ghost/Roles/Components/UseActorNameComponent.cs:
```diff
@@ -0,0 +1,10 @@
+/// This is used for...
+/// </summary>
+[RegisterComponent]
+public sealed class UseActorNameComponent : Component
```
а почему Актор а не Плеер

KIBORG04 replied:
Потому что так названа компонента, где лежит сикей

KIBORG04 replied:
Я изменил компоненту, щас вроде норм. Но вот про название незнаю

LudwigVonChesterfield commented on Content.Shared/Shuttles/Events/ShuttleConsoleChangeShipNameMessage.cs:
```diff
@@ -0,0 +1,12 @@
+namespace Content.Shared.Shuttles.Events;
+
+/// <summary>
+/// Raised on the client when it wishes to travel somewhere.
```
по названию сообщения звучит будто бы должно рейзиться когда хочешь поменять имя корабля, а не путешествовать?

KIBORG04 replied:
Копипаста

LudwigVonChesterfield commented on Resources/Prototypes/Entities/Markers/Spawners/ghost_roles.yml:
```diff
@@ -49,6 +49,24 @@
+- type: entity
+  id: SpawnPointShipEvent
+  name: ship event spawner
+  suffix: SHIPEVENT
```
Что такое суффикс? Буквально в чатик типо писать после имени?

KIBORG04 replied:
Это в консоле спавна вещей только, чтобы было удобнее

LudwigVonChesterfield commented on Resources/Prototypes/Entities/Mobs/Player/human.yml:
```diff
@@ -55,3 +55,43 @@
+      bloodlossHealDamage:
+        types:
+          Bloodloss: 0
+    - type: Respirator
```
а если не давать компоненту респирации и кровеносной системы, сможет ли терять кровь и задыхаться?

KIBORG04 replied:
Я незнаю, я вообще не шарю как оно работает и как сделать действительно бессмертную куклу, хотя вроде и копипащу

KIBORG04 replied:
> а если не давать компоненту респирации и кровеносной системы, сможет ли терять кровь и задыхаться?

И кстати, тут я не даю, а ПЕРЕОПРЕДЕЛЯЮ компоненты родителя.

Dushess0 commented on Content.Client/Shuttles/UI/ShuttleConsoleWindow.xaml:
```diff
@@ -11,6 +11,13 @@
+                    <Label Name="NameShipLabel" Text="{Loc 'shuttle-console-name-label'}" HorizontalAlignment="Center"/>
+                </controls:StripeBack>
+                <LineEdit Name="ShipName" Access="Public" HorizontalExpand="True"/>
+                <Button Name="ChangeNameShip" Access="Public" Text="Изменить"/>
```
Перевести надо будет

Dushess0 commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Launchers/launchers.yml:
```diff
@@ -119,6 +119,58 @@
   - type: Appearance
 
+- type: entity
+  name: Слабый рпг-пулемёт
```
а можно ли переводить названия предметов? Всмысле что бы у RU игроков было свое название, а у англ - свое?

KIBORG04 replied:
Вроде да, там есть перевод по энтити-тайпу, но это разве не чисто ивент для тау кеки?

KIBORG04 committed:
copypaste

KIBORG04 committed:
Заебись компонента теперь

KIBORG04 committed:
Реорганизация папок и пулемет со взрывными патронами

KIBORG04 committed:
Update projectiles.yml

KIBORG04 committed:
хз че сломалось

KIBORG04 commented:
Че ему надо

KIBORG04 committed:
Ивент для спавна шип-ебента.

Пиздец, savemap при сохранение у компоненты MetaData делает parent: 0, а savegrid делает parent: null. Как я понял, именно ЭТО изменение заставляет линтер НЕ спавнить в пушках патроны. Хз как я пофиксил, надеюсь линтер в CI не пошлёт.

KIBORG04 commented:
Я понятия не имею че надо этому УМЛ линтеру, но локально все работает

KIBORG04 committed:
Капец когда оно уже заработает

KIBORG04 committed:
???

KIBORG04 commented:
Короче кажется, если на мапе был какой-то объект, а потом объект как-то (не так хз) удаляешь, то файл кажись корраптится полностью

KIBORG04 committed:
допустим новая карта с нуля.............

KIBORG04 commented:
Я хз, у линтера мать сдохла кажись. Он новую карту не аппрувает, сделанную с нуля

KIBORG04 committed:
Merge branch 'master' of github.com:ThetaStation/Theta into shiptest-ebent

KIBORG04 committed:
?

KIBORG04 commented:
Бля, какая была вероятность, что когда мы решили создать тету и новые карты, то эти челики сломали сейвилку карт и её валидатор да так, что это еще и с запозданием дошло до ру билда.

KIBORG04 committed:
Ладно, новая карта..

KIBORG04 committed:
map

KIBORG04 committed:
забыл

Dushess0 committed:
syndicate fighter

Dushess0 commented:
добавил новый кораблик 
![image](https://user-images.githubusercontent.com/44201461/205369799-9b6c829a-5e80-4ebe-8ee5-61ad796acb4a.png)

Dushess0 committed:
beetle shuttle
